homeassistant/components/google_assistant/helpers.py

```diff
@@ -479,6 +503,10 @@ async def sync_serialize(self, agent_user_id):
             if area and area.name:
                 device["roomHint"] = area.name
 
+        device_info = await _get_device_info(self.hass, state.entity_id)
```

Comment:
Pass `dev_reg` and `ent_reg` to this function and `_get_area` above. That way we only fetch those once.

Reply:
I think we should avoid the entity and device lookup twice too. That is why I thought it might be better to do as a refactor.

Reply:
Done.